src/main/java/org/worshipsongs/importer/TopicDao.java

```diff
@@ -8,55 +8,60 @@
 /**
  * Created by Pitchu on 10/18/2015.
  */
-public class TopicDao {
-    public int getTopicId(Connection connection, String topic)
+public class TopicDao implements ITopicDao
+{
+    private Connection connection;
+    public TopicDao(Connection connection)
+    {
+        this.connection = connection;
+    }
+
+    public Topic getTopic(Topic topic)
```

Comment:
The method name and signature is misleading. The name says that this method returns a Topic, but actually this returns just the topic id. So rename this method as "getId". 
```
public int getId(String name)
{
}